pinot-common/src/main/java/org/apache/pinot/common/utils/tls/TlsUtils.java

```diff
@@ -227,19 +229,28 @@ public static void installDefaultSSLSocketFactory(String keyStoreType, String ke
       String trustStoreType, String trustStorePath, String trustStorePassword) {
     try {
       SecureRandom secureRandom = new SecureRandom();
-      SSLFactory sslFactory = RenewableTlsUtils.createSSLFactory(keyStoreType, keyStorePath, keyStorePassword,
-          trustStoreType, trustStorePath, trustStorePassword,
-          "SSL", secureRandom, true, false);
-      if (isKeyOrTrustStorePathNullOrHasFileScheme(keyStorePath)
-          && isKeyOrTrustStorePathNullOrHasFileScheme(trustStorePath)) {
-        RenewableTlsUtils.enableAutoRenewalFromFileStoreForSSLFactory(sslFactory, keyStoreType, keyStorePath,
-            keyStorePassword, trustStoreType, trustStorePath, trustStorePassword, "SSL", secureRandom,
-            PinotInsecureMode::isPinotInInsecureMode);
+      SSLContext sc;
+      if (keyStorePath == null && trustStorePath == null) {
+        // When neither keyStorePath nor trustStorePath is provided, a SSLFactory cannot be created. create SSLContext
+        // directly and use the default key manager and trust manager.
+        sc = SSLContext.getInstance(SSL_CONTEXT_PROTOCOL);
```

Comment:
Is this guaranteed to be not null?

Reply:
Yes, the protocol `SSL` is valid, so `SSLContext.getInstance(SSL_CONTEXT_PROTOCOL)` is guaranteed to be not null